include/pybind11/cast.h

```diff
@@ -434,11 +438,19 @@ struct string_caster {
     cast(const StringType &src, return_value_policy /* policy */, handle /* parent */) {
         const char *buffer = reinterpret_cast<const char *>(src.data());
         auto nbytes = ssize_t(src.size() * sizeof(CharT));
-        handle s = decode_utfN(buffer, nbytes);
-        if (!s) {
-            throw error_already_set();
+        if (std::is_same_v<CharT, uint8_t>
+#if defined(PYBIND11_CPP17)
+            or std::is_same_v<CharT, std::byte>
+#endif
+        ) {
+            return PyBytes_FromStringAndSize(buffer, nbytes);
+        } else {
```

Comment:
else-after-return is bad style and clang-tidy will complain when the CI is back up. Just remove the else block since the return terminates the function anyway.